pkg/http/api/v1/handler.go

```diff
@@ -67,7 +70,7 @@ func (h *requestHandler) AcceptScanRequest(res http.ResponseWriter, req *http.Re
 		return
 	}
 
-	scanResponse, err := h.scanner.Scan(scanRequest)
+	jobID, err := h.enqueuer.Enqueue(scanRequest)
```

Comment:
Maybe we need to avoid submit a duplicated useless scan job here by checking the previous status targeting the artifact contained in the scan request.

Reply:
I'd further optimise it in a separate PR. This one is already pretty big. Also comparing artifacts in scan request might led to some corner cases we have to consider, e.g. how to tell new scan request from rescanning just a few minutes later?

Reply:
Agree. This is just an enhancement suggestion. No big deal if we do not do it.